"Publish to Environment" is broken with option 'latest'

Something went wrong with PublishToEnvironment workflow recently. It was still working last week, and now not anymore.

If you run it with an option 'latest', it cannot find any artifacts. No setting been change on our side. 

```
Downloading BcContainerHelper preview version from Blob Storage
Import from C:\ProgramData\BcContainerHelper\6.0.6-preview1138\BcContainerHelper\BcContainerHelper.ps1
BcContainerHelper version 6.0.6-preview1138
Setting useCompilerFolder = True
BC.HelperFunctions emits usage statistics telemetry to Microsoft
Finding latest successful CICD run for branch main in repository sede-res/srs-bc
- https://api.github.com/repos/sede-res/srs-bc/actions/runs?per_page=100&page=1&exclude_pull_requests=true&status=completed&branch=main
Last successful CICD run for branch main in repository sede-res/srs-bc is 7932973460
Getting artifacts for workflow run 7932973460, mask Apps, projects SRS.nl,SRS.base and version *
No more artifacts found for workflow run 7932973460
Found 0 artifacts for mask Apps and projects SRS.nl SRS.base in workflow run 7932973460
Finding latest successful CICD run for branch main in repository sede-res/srs-bc
- https://api.github.com/repos/sede-res/srs-bc/actions/runs?per_page=100&page=1&exclude_pull_requests=true&status=completed&branch=main
Last successful CICD run for branch main in repository sede-res/srs-bc is 7932973460
Getting artifacts for workflow run 7932973460, mask Dependencies, projects SRS.nl,SRS.base and version *
No more artifacts found for workflow run 7932973460
Found 0 artifacts for mask Dependencies and projects SRS.nl SRS.base in workflow run 7932973460
AL-Go action ran: Deploy Telemetry Correlation Id: acb551a6-5f08-45fb-a136-01b004b9f2f7
Error: Unexpected error when running action. Error Message: Could not find any Apps artifacts for projects SRS.nl,SRS.base, version latest, StackTrace: at <ScriptBlock>, C:\a\_actions\microsoft\AL-Go\4c5bfbca1adebbf997f63882df4b9074a19aac1d\Actions\Deploy\Deploy.ps1: line 142 <- at <ScriptBlock>, C:\a\_temp\f2d4a[40](https://github.com/sede-res/srs-bc/actions/runs/7964374281/job/21741795233#step:6:42)2-7cd2-[44](https://github.com/sede-res/srs-bc/actions/runs/7964374281/job/21741795233#step:6:46)03-83b5-7bf5e2b8ebf9.ps1: line 4 <- at <ScriptBlock>, <No file>: line 1
Error: Process completed with exit code 1.
```

If, i navigate to that specific workflow i can see it ran successfully and produced artifacts

![image](https://github.com/microsoft/AL-Go/assets/16342128/2f39155e-e404-40c6-b0f0-098c20f81ada)

If, i specify exact version '1.0.1292.0' instead of 'latest' it does work.

I tried to update to the latest preview version. It's the same result.

--- a/Actions/Github-Helper.psm1
+++ b/Actions/Github-Helper.psm1
@@ -884,7 +884,7 @@ function GetArtifactsFromWorkflowRun {
     $page = 1
 
     # Get sanitized project names (the way they appear in the artifact names)
-    $projects = @(@($projects.Split(',')) | ForEach-Object { $_.Replace('\','_').Replace('/','_') })
+    $projectArr = @(@($projects.Split(',')) | ForEach-Object { $_.Replace('\','_').Replace('/','_') })
 
     # Get the artifacts from the the workflow run
     while($true) {
@@ -897,7 +897,7 @@ function GetArtifactsFromWorkflowRun {
             break
         }
 
-        foreach($project in $projects) {
+        foreach($project in $projectArr) {
             $artifactPattern = "$project-*-$mask-*" # e.g. "MyProject-*-Apps-*", format is: "project-branch-mask-version"
             $matchingArtifacts = @($artifacts.artifacts | Where-Object { $_.name -like $artifactPattern })
 
@@ -922,7 +922,7 @@ function GetArtifactsFromWorkflowRun {
         $page += 1
     }
 
-    Write-Host "Found $($foundArtifacts.Count) artifacts for mask $mask and projects $($projects -join ',') in workflow run $workflowRun"
+    Write-Host "Found $($foundArtifacts.Count) artifacts for mask $mask and projects $($projectArr -join ',') in workflow run $workflowRun"
 
     return $foundArtifacts
 }

--- a/RELEASENOTES.md
+++ b/RELEASENOTES.md
@@ -2,6 +2,9 @@
 
 Note that when using the preview version of AL-Go for GitHub, we recommend you Update your AL-Go system files, as soon as possible when informed that an update is available.
 
+### Issues
+- Issue 940 Publish to Environment is broken when specifying projects to publish
+
 ### New Settings
 - `UpdateALGoSystemFilesEnvironment`: The name of the environment that is referenced in job `UpdateALGoSystemFiles` in the _Update AL-Go System Files_ workflow. See [jobs.<job_id>.environment](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idenvironment) for more information. Currently, only setting the environment name is supported.
 

